### PR TITLE
refactor: pyproject.toml to poetry

### PR DESCRIPTION
**Pull request by MyuddinKhatri**
Issue - https://github.com/agritheory/cloud_storage/issues/61
- Refactor pyproject.toml to poetry
- Move test dependencies to dev dependencies section

Changes collected in test_utils [test_utils](https://github.com/agritheory/test_utils)

- [x] .github/workflows/validate_customizations.py

### CODE REVIEW

**Comment by MyuddinKhatri** [comment]
@agritheory I have added [version](https://github.com/agritheory/cloud_storage/pull/65/files#diff-50c86b7ed8ac2cf95bd48334961bf0530cdc77b5a56f852c5c61b89d735fd711R3) from __init__.py, as it is required for package mode, the versioning can be made dynamic with these - 
- https://github.com/mtkennerly/poetry-dynamic-versioning
- https://github.com/tiangolo/poetry-version-plugin

I would like to know your thought should we add these or keep version as it is in pyproject.toml?

**Comment by agritheory** [comment]
@MyuddinKhatri How is this different (in outcome) from how we've implemented python semantic version?

**Comment by MyuddinKhatri** [comment]
> @MyuddinKhatri How is this different (in outcome) from how we've implemented python semantic version?

@agritheory I got confused between the existing dynamic = ["version"] via flit_core, I wanted to include version in current pyproject.toml with poetry as version is mandatory, and I wanted to do something similar to this, but forgot about the python semantic version.

File: pyproject.toml [hunk 1]
```diff
@@ -1,24 +1,23 @@
-[project]
+[tool.poetry]
 name = "cloud_storage"
-authors = [
-    { name = "AgriTheory", email = "support@agritheory.com"}
-]
+version = "14.0.0"
 description = "Frappe App for integrating with cloud storage applications"
-requires-python = ">=3.8"
+authors = ["AgriTheory <support@agritheory.com>"]
```
Comment (agritheory) [inline pyproject.toml]: This should be support@agritheory.dev for now.

**Review by agritheory — CHANGES_REQUESTED** [review]
Can you rebase this to version-15?

**Comment by agritheory** [comment]
@MyuddinKhatri Can you fix the linting here?

**Comment by MyuddinKhatri** [comment]
> @MyuddinKhatri Can you fix the linting here?

Sure, I'll look into it why linting is failing.

**Comment by agritheory** [comment]
@MyuddinKhatri I recognize the problem and just pushed a fix

**Review by agritheory — APPROVED** [review]

**Pull request merged**